Source/JavaScriptCore/runtime/JSCJSValue.h

```diff
@@ -554,42 +558,22 @@ class JSValue {
     EncodedValueDescriptor u;
 };
 
-#if USE(JSVALUE32_64)
-struct OrderedHashTableTraits {
-    ALWAYS_INLINE static void set(JSValue* value, uint32_t number)
-    {
-        value->u.asBits.tag = JSValue::Int32Tag;
-        value->u.asBits.payload = number;
-    }
-    ALWAYS_INLINE static void increment(JSValue* value)
-    {
-        ASSERT(value->isInt32());
-        value->u.asBits.payload++;
-    }
-    ALWAYS_INLINE static void decrement(JSValue* value)
-    {
-        ASSERT(value->isInt32());
-        value->u.asBits.payload--;
-    }
-};
-#else
 struct OrderedHashTableTraits {
     ALWAYS_INLINE static void set(JSValue* value, uint32_t number)
     {
-        value->u.asInt64 = JSValue::NumberTag | number;
+        *value = JSValue(static_cast<int32_t>(number));
```

Comment:
I think this should be landed in a separate PR, just in case it causes a perf regression